pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
#25787 seems to have broken the ability to make the type assertions these checks rely on[0] since the `runtime.Object` is now a `rest.UpdatedObjectInfo` which hides the underlying object being updated to.

Would it be reasonable to pull the privilege escalation checks from this PR and revisit a more robust multilevel security system for beta? Based on the conversation here (and the above issue) it seems like there might be a better home for these checks anyway.

cc @erictune 

[0] https://github.com/kubernetes/kubernetes/pull/25634/files#diff-0ddbb384845118d5b907e3498b76455cR66